roles/nginxplus/tasks/plus/install-plus.yml

```diff
@@ -1,13 +1,37 @@
 ---
-- import_tasks: setup-license.yml
+- name: nginxplus | set up license
+  ansible.builtin.import_tasks: setup-license.yml
 
-- import_tasks: setup-debian.yml
+- name: nginxplus | set up ubuntu
+  ansible.builtin.import_tasks: setup-debian.yml
 
-- name: "Install: Install NGINX Plus"
-  apt:
-    name: nginx-plus
+- name: nginxplus |  NGINX Plus license verification
+  ansible.builtin.blockinfile:
+    path: /etc/apt/apt.conf.d/90nginx
+    create: true
+    block: |
+      Acquire::https::pkgs.nginx.com::Verify-Peer "true";
+      Acquire::https::pkgs.nginx.com::Verify-Host "true";
+      Acquire::https::pkgs.nginx.com::SslCert     "/etc/ssl/nginx/nginx-repo.crt";
+      Acquire::https::pkgs.nginx.com::SslKey      "/etc/ssl/nginx/nginx-repo.key";
     state: present
-  notify: Start NGINX
+    mode: 0444
 
-- import_tasks: delete-license.yml
-  when: nginx_delete_license
+- name: nginxplus | NGINX Plus repository
+  ansible.builtin.apt_repository:
+    filename: nginx-plus
+    repo: deb {{ nginx_signed_statement }} https://pkgs.nginx.com/plus/ubuntu {{ ansible_distribution_release }} nginx-plus
+    update_cache: false
+    state: present
+    mode: 0644
+  when: nginx_manage_repo | bool
+
+- name: nginxplus | {{ nginx_setup | capitalize }} NGINX Plus
```

Comment:
This is a cool construction.

In `group_vars/common.yml` I see `nginx_setup: install` - what other values could `nginx_setup` have? Would we set it to `upgrade` or `downgrade` somehow when changing the version? I'm trying to envision what this output will look like.